packages/core/upload/server/graphql.js

```diff
@@ -98,12 +97,29 @@ module.exports = ({ strapi }) => {
           },
 
           async resolve(parent, args) {
-            const { file: upload, info, ...fields } = args;
-
-            const file = await formatFile(upload, info, fields);
-            const uploadedFile = await getUploadService('upload').uploadFileAndPersist(file);
+            // create temporary folder to store files for stream manipulation
+            const tmpFolderPath = await fse.mkdtemp(path.join(os.tmpdir(), 'strapi-upload-'));
```

Comment:
it feels heavy to create a temp folder and pass it around everywhere. Can't we create tmp dirs on the fly to make the code simpler? or set a tmpDir info on the file descriptor